InfiniteRecharge/src/main/java/frc/robot/Robot.java

```diff
@@ -117,6 +121,88 @@ public void robotInit() {
+                driveSubsystem)
+            );
+        }
+
+        //////////////////////////////////////////////
+        // Intake Subsystem
+        if (config.enableIntakeSubsystem) {
+            b.operatorToggleIntake.whenPressed(new InstantCommand(() -> intakeSubsystem.toggleIntakeArm(), intakeSubsystem));
+
+            b.operatorIntake.whenHeld(new InstantCommand(() -> intakeSubsystem.intake(), intakeSubsystem)).whenReleased(new InstantCommand(() -> intakeSubsystem.off(), intakeSubsystem));
+            b.operatorOutake.whenHeld(new InstantCommand(() -> intakeSubsystem.outake(), intakeSubsystem)).whenReleased(new InstantCommand(() -> intakeSubsystem.off(), intakeSubsystem));
+        }
+
+        //////////////////////////////////////////////
+        // Climb Subsystem
+        if (config.enableClimbSubsystem) {
+            b.operatorActivateClimb.and(b.driveActivateClimb).whenActive(new InstantCommand(() -> climbSubsystem.toggleActive(), climbSubsystem));
+
+            // continuously call moveArms() with the axis valuse
+            // moveArms() will apprpriately call pitRewind or manualClimb
+            climbSubsystem.setDefaultCommand(new RunCommand(
+                () -> climbSubsystem.moveArms(
+                    b.operatorControl.getRawAxis(b.operatorClimbLeft), 
+                    b.operatorControl.getRawAxis(b.operatorClimbRight)), 
+                climbSubsystem)
+            );
+            
+        }
+
+
+        //////////////////////////////////////////////
+        // Shooter Subsystem
+        if (config.enableShooterSubsystem) {
+            b.operatorAimBot.whenHeld(new InstantCommand(() -> shooterSubsystem.autoAim(), shooterSubsystem)).whenReleased(new InstantCommand(() -> shooterSubsystem.stopAutoAim(), shooterSubsystem));
+            b.operatorSpinUp.whenHeld(new InstantCommand(() -> shooterSubsystem.startSpinningUp(), shooterSubsystem)).whenReleased(new InstantCommand(() -> shooterSubsystem.stopSpinningUp(), shooterSubsystem));
+            b.operatorFeeder.whenHeld(new InstantCommand(() -> shooterSubsystem.spinFeeder(), shooterSubsystem)).whenReleased(new InstantCommand(() -> shooterSubsystem.stopSpinningFeeder(), shooterSubsystem));
+            b.operatorFire.whenHeld(new InstantCommand(() -> shooterSubsystem.spinBMS(), shooterSubsystem)).whenReleased(new InstantCommand(() -> shooterSubsystem.holdFire(), shooterSubsystem));
+
+            shooterSubsystem.setDefaultCommand(new RunCommand(
+                () -> shooterSubsystem.rotate(
+                    Math.abs(b.operatorControl.getRawAxis(b.operatorAzimuth)), 
+                    Math.abs(b.operatorControl.getRawAxis(b.operatorElevation))),
+                shooterSubsystem)
+            );
+
+            b.operatorNextElevation.whenPressed(new InstantCommand(() -> shooterSubsystem.nextPositionElevation(), shooterSubsystem)).whenReleased(new InstantCommand(() -> shooterSubsystem.resetPositionElevationSwitcher(), shooterSubsystem));

```

Comment:
Won't this only queue up the latter command? whenPressed and whenReleased will happen on the same loop I think?

Reply:
I think we want a StartEndCommand that calls nextPositionElevation for the start and resetPositionElevationSwitcher for the finish

Reply:
Why not just the call both the function right after each other in an InstantCommand?